### PR TITLE
pythonPackages.paho-mqtt: 1.1 -> 1.4.0

### PR DESCRIPTION
**Pull request by andir**
###### Motivation for this change

Updated to the latest stable release.

cc @mogorman 

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by flokli** [comment]
@GrahamcOfBorg build python3Packages.paho-mqtt python2Packages.paho-mqtt

**Comment by flokli** [comment]
tests fail on darwin. could you have a look?

File: pkgs/development/python-modules/paho-mqtt/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+  # fetch from GitHub so we can run the tests
+  src = fetchFromGitHub {
+    owner = "eclipse";
+    repo = "paho.mqtt.python";
+    rev = "v${version}";
+    sha256 = "1xg9ppz2lqacd9prsrx93q2wfkjjyla03xlfw74aj1alz9ki5hrs";
+  };
+
+  buildInputs = [ pylama ];
```
Comment (FRidh) [inline pkgs/development/python-modules/paho-mqtt/default.nix]: I suppose this is used for tests, or not? If so, it should be in checkInputs. Also, I think we can avoid including it altogether as it doesn't really make much sense having it running in our build.

Comment (flokli) [reply pkgs/development/python-modules/paho-mqtt/default.nix]: pylama is a linter - right, we don't necessarily need to lint the codebase, although I'd still suggest to keep running the unit tests ;-)

**Pull request closed without merging**